Modules/StudyProgramme/classes/class.ilObjStudyProgramme.php

```diff
@@ -163,7 +134,7 @@ protected function readSettings() {
 		if (!$id) {
 			throw new ilException("ilObjStudyProgramme::loadSettings: no id.");
 		}
-		$this->settings = $this->getSettingsRepository()->read($this->getId());
+		$this->settings = ilStudyProgrammeDIC::dic()['model.Settings.ilStudyProgrammeSettingsRepository']->read($this->getId());
```

Comment:
Zieh diese Sachen doch bitte schon im Konstruktor aus dem DIC in Properties. Dann ist zumindestens das "innere" der Klasse frei von Service Locator.

Reply:
Ok, aber das macht die Performancevorteile der late initiallization zunichte...